crates/engine/src/rocksdb_engine.rs

```diff
@@ -99,6 +103,29 @@ impl RocksEngine {
         }
         Ok(size)
     }
+
+    /// Apply snapshot from file
+    pub async fn apply_snapshot_from_file<P>(
+        &self,
+        snap_path: P,
+        tables: &[&'static str],
+    ) -> Result<(), EngineError>
+    where
+        P: AsRef<Path>,
+    {
+        let mut snapshot_f = tokio::fs::File::open(snap_path).await?;
+        let tmp_path = temp_dir().join(format!("snapshot-{}", uuid::Uuid::new_v4()));
+        let mut rocks_snapshot = RocksSnapshot::new_for_receiving(tmp_path)?;
+        let mut buf = BytesMut::with_capacity(SNAPSHOT_CHUNK_SIZE);
+        while let Ok(n) = read_buf(&mut snapshot_f, &mut buf).await {
+            if n == 0 {
+                break;
+            }
+            rocks_snapshot.write_all(buf.split().freeze()).await?;
+        }
+        self.apply_snapshot(rocks_snapshot, tables).await?;
+        Ok(())
```

Comment:
Why should convert a snapshot file into a temp file and then apply it? Can we directly use the `snapshot_f` to construct the `RocksSnapshot` instance?

Reply:
snapshot_f is a single file, RocksSnapshot a set of sst files, the snapshot must be separated into multiple sst files before it can be applied to the RocksDB instance.